ledger-service/http-json-cli/src/main/scala/com/daml/http/OptionParser.scala

```diff
@@ -4,17 +4,19 @@
 package com.daml.http
 
 import java.nio.file.Paths
-
 import com.daml.ledger.api.tls.TlsConfigurationCli
-import com.typesafe.scalalogging.StrictLogging
+import com.daml.logging.DeferringLogger
 import scopt.{Read, RenderingMode}
 
 import scala.concurrent.duration.{Duration, FiniteDuration}
 import scala.util.Try
 
 class OptionParser(getEnvVar: String => Option[String], supportedJdbcDriverNames: Set[String])
-    extends scopt.OptionParser[Config]("http-json-binary")
-    with StrictLogging {
+    extends scopt.OptionParser[Config]("http-json-binary") {
+
+  // We need a deferring logger because we do not know
+  // whether the user expects json or plain logging output.
+  val logger: DeferringLogger = DeferringLogger(getClass)
```

Comment:
this seems like a lot of complexity and slightly weird UX (by delaying logs) just to get the right format for logs from the config parser. I’m not sure we actually log anything useful here apart from maybe some deprecation warnings here and there which we could just move out of this.

I would suggest to drop this and just accept that for this PR logs will have the wrong format and move logs out of option parsing in a separate PR to fix that properly.

Reply:
It definitely is a bit weird to defer the logs. I aimed to be consistent here, however if you think this is unnecessary then I can drop the DeferringLogger.